boards/arm/hexiwear_k64/Kconfig.defconfig

```diff
@@ -22,23 +22,13 @@ config MCG_VDIV0
 config MCG_FCRDIV
 	default 1
 
-if UART_MCUX
-
-config UART_MCUX_0
-	default y if UART_CONSOLE
-
-config UART_MCUX_4
-	default y if BT_UART
-
-endif # UART_MCUX
-
 if PINMUX_MCUX
 
 config PINMUX_MCUX_PORTA
 	default y if MAX30101
 
 config PINMUX_MCUX_PORTB
-	default y if UART_MCUX_0
+	default y if "$(dt_dev_exist_by_path,/soc/uart@4006a000)"
```

Comment:
Is there a way to reach the node by its label or at least node-name? (with node-name or label as follows `[label:] node-name[@unit-address]`)

Reply:
It would be good to have the possibility of referring to the node by its label. For instance for SoCs 
 with Secure and Non-Secure domains, where the same device appears under different addresses, depending on the domain.

Reply:
I'm not a fan of using the label (i.e., the token that can be use as `&label`) because it's just an helper for the canonical identifier (path) and is used only in the source representation of the device tree structure.  I acknowledge that's being a stickler, but the feature is also under-defined: are the labels for nodes and properties in the same or distinct namespaces?  what is the behavior if a label is multiply defined?  what is the interpretation of a label within a property value?

I'd support using the `zephyr,name` property that might replace the `label` property.

The case of two addresses for the same device in different domains could be handled by checking for both addresses (or device names).

Reply:
I don't think `label` makes sense either.  I device name should go away or not be required in the future, we should have APIs like:

`device_get_binding_by_dt_path(path)`
`device_get_binding_by_dt_alias(alias)`

these can get resolved at build time in the future when we do code generation for the driver.

`label` and even `zephyr,name` shouldn't be required.

Reply:
> It would be good to have the possibility of referring to the node by its label. For instance, for SoCs with Secure and Non-Secure domains, where the same device appears under different addresses, depending on the domain.

Can you give an example of what you are needing for secure v non-secure.

Reply:
@galak

> device_get_binding_by_dt_path(path)
device_get_binding_by_dt_alias(alias)

I think most of us now see that alias should be used in quite specific cases only.
So we're left with `device_get_binding_by_dt_path(path)`, which get to use base address each time we need to bind a device. Can you elaborate why being able to call a device by a human friendly name (whatever way we find to get it) doesn't make sense?

Reply:
> > It would be good to have the possibility of referring to the node by its label. For instance, for SoCs with Secure and Non-Secure domains, where the same device appears under different addresses, depending on the domain.
> 
> Can you give an example of what you are needing for secure v non-secure.

I've got nothing specific at hand now. But I thought about a case like this one, when a Kconfig setting is dependent on specific peripheral instance being enabled. Then we'll need to call `dt_dev_exist_by_path` twice, for secure and non-secure address.

Reply:
> I've got nothing specific at hand now. But I thought about a case like this one, when a Kconfig setting is dependent on specific peripheral instance being enabled. Then we'll need to call `dt_dev_exist_by_path` twice, for secure and non-secure address.

Why twice?  When we build Zephyr we'd be targeting secure or non-secure, and that DTS for that target is what we'd utilize.  Its possible also that we could have something like:

```
Kconfig foo
	depends on SECURE && dt_dev_exist_by_path("/uart@f000")
```

Reply:
>  The problem is what consistent friendly name do we define that works across all boards, arch's, SoC's, etc?

So, I'm not sure why this should have to be consistent across all of these. And what we gain to have this if we lose a big deal which is ease of use in a single board, which will be the most common use case.
Introducing the Secure and Non-Secure ranges in device tree to deal easily with Secure and Non-Secure domain have been a great help in C-M33 developments. I have been able to compare vs project not having this tool, and then which lost a lot of time dealing with establishing devices ate the correct address depending on the secure/non secure context. Having not to deal with addresses thanks to use of range was a great time saving in this context.
Having to deal with base addresses for basic device manipulation is error prone going very much backwards in ease of use for most people (how much time will we spend in counting 0's?).

Reply:
> So, I'm not sure why this should have to be consistent across all of these. And what we gain to have this if we lose a big deal which is ease of use in a single board, which will be the most common use case.
> Introducing the Secure and Non-Secure ranges in device tree to deal easily with Secure and Non-Secure domain have been a great help in C-M33 developments. I have been able to compare vs project not having this tool, and then which lost a lot of time dealing with establishing devices ate the correct address depending on the secure/non secure context. Having not to deal with addresses thanks to use of range was a great time saving in this context.
> Having to deal with base addresses for basic device manipulation is error prone going very much backwards in ease of use for most people (how much time will we spend in counting 0's?).

But in which context, the use of the DT_NXP_KINETIS_UART_4006D000 is limited to the cases that we need to know a unique device instance, right now that is pinmux.  Eventually this should be coming from DT anywhere so it would go away.

I'm concerned that we have a consistent solution across all of Zephyr, and not something that varies from SoC to SoC.

Reply:
> right now that is pinmux

There are other occurences here and there, where there is dedicated behavior for particular instances, right now I can see at least UART and I2C on STM32

Reply:
> There are other occurences here and there, where there is dedicated behavior for particular instances, right now I can see at least UART and I2C on STM32

what other examples beyond pinmux do you have?

Reply:
> what other examples beyond pinmux do you have?

I meant in I2C and UART STM32 drivers there is some code specific to some particular device instances on which we call particular functions because these instances are somewhat different from the other ones (ie not pinmux related)

Reply:
> I meant in I2C and UART STM32 drivers there is some code specific to some particular device instances on which we call particular functions because these instances are somewhat different from the other ones (ie not pinmux related)

Can you point me to file/lines to see that.  How did you see this working w/codegen?

Reply:
> So one way as a middle ground would be to leave the per device Kconfig symbols as such:
> 
> ```
> if UART_MCUX
> 
> config UART_MCUX_0
>    bool
>    depends on "$(dt_dev_exist_by_path,/soc/uart@4006a000)"
> 
> config UART_MCUX_1
>    bool
>    depends on "$(dt_dev_exist_by_path,/soc/uart@4006b000)"
> 
> ...
> ```

This assumes that UART_MCUX_0 will always have the same base address across different SoCs, which is not necessarily true.

Reply:
> This assumes that UART_MCUX_0 will always have the same base address across different SoCs, which is not necessarily true.

We can always do:

```
   depends on "$(dt_dev_exist_by_path,/soc/uart@4006a000)" if SOC_SERIES_KINETIS_K6X
   depends on "$(dt_dev_exist_by_path,/soc/uart@fffef000)" if SOC_SERIES_KINETIS_K1X
```

Reply:
> Can you point me to file/lines to see that. How did you see this working w/codegen?

I2C: https://github.com/zephyrproject-rtos/zephyr/blob/master/drivers/i2c/i2c_ll_stm32.c#L200 and following
UART: https://github.com/zephyrproject-rtos/zephyr/blob/635191ac8c563990abb9d471aed0d97a4b86cc95/drivers/serial/uart_stm32.c#L80

In the codegen solution I proposed, these flags were kept (generated using dts `label` when required).

Reply:
Gotcha, so I think we should introduce something like "soc-id" as a property.  With the definition is that its a 'string' that relates to the data sheet / manual / docs of the SoC.

Reply:
> Gotcha, so I think we should introduce something like "soc-id" as a property. With the definition is that its a 'string' that relates to the data sheet / manual / docs of the SoC.

Maybe `soc-label`.

Reply:
> Maybe `soc-label`.

Which differs how from `label` (today) or `zephyr,name` (one of many proposed alternatives)?  Other than the current use of the property is a temporary workaround for incomplete tooling and may disappear, but the new use would make it permanent?  Do we need two properties with materially the same content?

Reply:
> Which differs how from `label` (today) or `zephyr,name` (one of many proposed alternatives)? Other than the current use of the property is a temporary workaround for incomplete tooling and may disappear, but the new use would make it permanent? Do we need two properties with materially the same content?

We'll its not that different from `'zephyr,name'`, just that we are defining its intent more clearly.  As you've stated `label` shouldn't have been used, and my hope is with codegen it goes away since we can reference a device by path or alias.  However we do seem to want/need something that relates to how SoC's document their IP.  So `'soc-label'` is an attempt to provide that.  Its shouldn't be used outside of a soc, and meant specifically for code generation purposes so what gets generated is relatable to manual/docs for the SoC.

Reply:
Based on discussion I opened up #12546 for supporting 'soc-label'.